nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/trash/test/TestTrashBackwardMechanism.java

```diff
@@ -120,12 +116,10 @@ public void testMechanismTriggersOneListener() {
 
     public static class BulkListenersFilter implements LogCaptureFeature.Filter {
 
-        protected final Set<String> bulkListenerNames = Stream.of(BulkLifeCycleChangeListener.class,
-                BulkTrashedStateChangeListener.class).map(Class::getName).collect(Collectors.toSet());
-
         @Override
         public boolean accept(LoggingEvent event) {
-            return Level.DEBUG.equals(event.getLevel()) && bulkListenerNames.contains(event.getLoggerName());
+            return Level.DEBUG.equals(event.getLevel())
+                    && event.getLoggerName().equals(BulkLifeCycleChangeListener.class.getName());
```

Comment:
Filter can now be converted to `FilterOn` annotation.

Reply:
Also you're changing logic of test which asserted that only one recursive mechanism was trigger.